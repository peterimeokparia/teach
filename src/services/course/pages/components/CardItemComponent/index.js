import Card from '@mui/material/Card';
import CardContent from '@mui/material/CardContent';

const CardItemComponent = ({ 
    cardItemCollection, 
    handleClickedCardItem, 
    handleCardItemOnBlur, 
    children, 
    cardItemProps 
}) => { 
    return (( cardItemCollection )?.map((item, index) => ( 
                    <div className='col' onClick={() => handleClickedCardItem( item )}> 
                    {  <div className="cardItem">
                        { <Card sx={{ 
                                minWidth: 275, 
                                maxHeight:305, 
                                backgroundColor: item?.color, 
                                marginBottom: 3.6, 
                                boxShadow: 9 
                            }}  onBlur={() => handleCardItemOnBlur(item)}>
                                <CardContent>
                                    { children( item, index, cardItemProps ) }
                                </CardContent>
                            </Card>  
                        }
                        </div>
                    }
                    </div>
                )
            ));
};

export default CardItemComponent;

 