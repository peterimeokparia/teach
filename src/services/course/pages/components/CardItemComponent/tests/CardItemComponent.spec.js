import React from 'react';
import CardItemComponent  from 'services/course/pages/components/CardItemComponent';
import renderer from 'react-test-renderer';

describe('CardItemComponent', () =>  {  
    let collection = [ {title: 'testOO1'}, {title: 'test002'}];
    const handleClickedCarditem = jest.fn();
    const handleCardItemOnBlur = jest.fn();
    const cardItemProps = {};

  it('renders as expected', async () => {
    const tree = renderer.create(
        <CardItemComponent
          cardItemCollection={collection}
          handleClickedCardItem={handleClickedCarditem}
          handleCardItemOnBlur={handleCardItemOnBlur}
          cardItemProps={cardItemProps}
        >
            {( cardItem, index, cardItemProps ) => ( <div> { cardItem?.title } </div>)

            }
        </CardItemComponent> 
    );
      expect(tree.toJSON()).toMatchSnapshot();
  })
});




