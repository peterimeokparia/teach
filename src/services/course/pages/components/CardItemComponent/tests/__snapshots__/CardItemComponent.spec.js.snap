// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardItemComponent renders as expected 1`] = `
Array [
  <div
    className="col"
    onClick={[Function]}
  >
    <div
      className="cardItem"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1emxec4-MuiPaper-root-MuiCard-root"
        onBlur={[Function]}
      >
        <div
          className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <div>
             
            testOO1
             
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="col"
    onClick={[Function]}
  >
    <div
      className="cardItem"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1emxec4-MuiPaper-root-MuiCard-root"
        onBlur={[Function]}
      >
        <div
          className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <div>
             
            test002
             
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
